app/src/main/java/io/homeassistant/companion/android/controls/ClimateControl.kt

```diff
@@ -35,6 +36,7 @@ class ClimateControl {
                 )
             )
             control.setTitle((entity.attributes["friendly_name"] ?: entity.entityId) as CharSequence)
+            control.setSubtitle(area?.name ?: "")
```

Comment:
You have added this identical line to every single control class. That very much suggests to me that it should be done in the superclass that all controls inherit from.

Actually, the existing code is giving a poor example to follow, since it also sets the title using the same code repeated in every child class, which again should be handled by the superclass.

Reply:
I agree with you but decided to follow the existing code style to keep changes in this PR to a minimum (it is already suffering from feature creep in my opinion in WebSocketRepositoryImpl). If merged I can submit another PR for this?

Reply:
Lets make sure to keep our PRs down to a minimum as it helps make the review process easier. It can be very easy to keep finding things to fix but it doesn't mean it should all be done in one PR. The smaller the PR the easier it is to get reviewed and merged.

Reply:
My point was that doing it in the superclass probably would have made the PR smaller - just do it once, not have to add it separately to 11 classes.  (Fixing the existing repeated code to set the title I agree is going beyond the scope of this PR).  More than happy that these can be addressed in a follow on PR.

Reply:
Perhaps it would've saved a line or two of code in the end, but we're talking about 9 new lines of code which are easy to review whereas @zmarties' suggestion will require changing the interface to a class, updating all controls accordingly, and checking everything to see whether it is unique to the control or not and should go the the HaControl class.
Marking this as resolved, improvements to go in a separate PR.